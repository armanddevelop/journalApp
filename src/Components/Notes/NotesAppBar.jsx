import { useDispatch, useSelector } from "react-redux";
import {
  startSaveNoteAction,
  startUploadImageAction,
} from "../../Actions/notes";

export const NotesAppBar = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector(({ notes }) => notes);
  const handleSaveNote = () => {
    dispatch(startSaveNoteAction(note));
  };
  const handlePictureClick = () => {
    document.getElementById("fileSelector").click();
  };
  const handleFileChange = ({ target }) => {
    const file = target.files[0];
    if (!file) return;
    dispatch(startUploadImageAction(file));
    document.getElementById("fileSelector").value = "";
  };

  return (
    <div className="notes__appbar">
      <span>19/03/2022</span>
      <input
        type="file"
        id="fileSelector"
        name="picture"
        style={{ display: "none" }}
        onChange={(e) => handleFileChange(e)}
      />
      <div>
        <button className="btn" onClick={handlePictureClick}>
          Picture
        </button>
        <button className="btn" onClick={handleSaveNote}>
          Save
        </button>
      </div>
    </div>
  );
};
