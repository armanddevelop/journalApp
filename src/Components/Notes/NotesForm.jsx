import { useEffect, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";
import { notesActiveAction, startDeleteNoteAction } from "../../Actions/notes";
import { useForm } from "../../Hooks/useForm";

export const NotesForm = () => {
  const { active: note } = useSelector(({ notes }) => notes);
  const dispatch = useDispatch();
  const { values, handleInputChange, reset } = useForm(note);
  const { title, body } = values;
  const idActive = useRef(note.id);

  useEffect(() => {
    if (idActive.current !== note.id) {
      reset(note);
      idActive.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    const { id, date, title, body, url } = values;
    const noteObj = {
      date,
      title,
      body,
      url,
    };
    dispatch(notesActiveAction(id, noteObj));
  }, [values, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleteNoteAction(idActive.current));
  };

  return (
    <div className="notes__content">
      <input
        type="text"
        placeholder="Some awsome title"
        className="notes__title-input"
        autoComplete="off"
        autoFocus={true}
        name="title"
        value={title}
        onChange={(e) => handleInputChange(e)}
      />
      <textarea
        placeholder="What are you doing today"
        className="notes__textarea"
        name="body"
        value={body}
        onChange={(e) => handleInputChange(e)}
      ></textarea>
      <button className="btn btn-danger" onClick={handleDelete}>
        Delete note
      </button>
      {note.url && (
        <div className="notes__image">
          <img src={note.url} alt="source" />
        </div>
      )}
    </div>
  );
};
