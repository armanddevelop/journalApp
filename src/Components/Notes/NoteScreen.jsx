import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          placeholder="Some awsome title"
          className="notes__title-input"
          autoComplete="off"
        />
        <textarea
          placeholder="What are you doing today"
          className="notes__textarea"
        ></textarea>
        <div className="notes__image">
          <img
            src="https://assets.leevalley.com/Size5/10061/49L0795-everyman-s-journal-u-02-r.jpg"
            alt="source"
          />
        </div>
      </div>
    </div>
  );
};
