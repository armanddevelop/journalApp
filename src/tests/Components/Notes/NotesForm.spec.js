import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {
  startDeleteNoteAction,
  notesActiveAction,
} from "../../../Actions/notes";
import { NotesForm } from "../../../Components/Notes/NotesForm";

jest.mock("../../../Actions/notes", () => ({
  startDeleteNoteAction: jest.fn(),
  notesActiveAction: jest.fn(),
}));
const initState = {
  auth: {
    name: "Alicha",
  },
  ui: {
    errorMsgFb: {
      error: false,
      msgError: "",
    },
  },
  notes: {
    active: {
      id: "safdbdg32435",
      title: "Hi note",
      body: "hi lich",
      date: 1649197509169,
    },
    notes: [],
    saveNote: {
      isSave: true,
      message: "All rigth",
      title: "note save",
    },
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore(initState);
store.dispatch = jest.fn();

describe("Test in NotesForm Component", () => {
  const wrapper = mount(
    <Provider store={store}>
      <NotesForm />
    </Provider>
  );

  it("SnapShot to NotesForm component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should execute startDeleteNoteAction", () => {
    wrapper.find(".notes__title-input").simulate("change", {
      target: {
        name: "title",
        value: "Bye note",
      },
    });
    const { notes } = initState;

    expect(notesActiveAction).toHaveBeenCalledWith(notes.active.id, {
      body: "hi lich",
      date: 1649197509169,
      title: "Bye note",
      url: undefined,
    });
  });

  it("Should excute startDeleteNoteAction", () => {
    wrapper.find(".btn").prop("onClick")();
    const { notes } = initState;

    expect(startDeleteNoteAction).toHaveBeenCalledWith(notes.active.id);
  });

  it("Should not exist url img div", () => {
    expect(wrapper.find(".notes__image").exists()).toBe(false);
  });

  it("Should exist url img div", () => {
    const initState = {
      auth: {
        name: "Alicha",
      },
      ui: {
        errorMsgFb: {
          error: false,
          msgError: "",
        },
      },
      notes: {
        active: {
          id: "safdbdg32435",
          title: "Hi note",
          body: "hi lich",
          date: 1649197509169,
          url: "https//:www.img.com",
        },
        notes: [],
        saveNote: {
          isSave: true,
          message: "All rigth",
          title: "note save",
        },
      },
    };
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    let store = mockStore(initState);
    store.dispatch = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <NotesForm />
      </Provider>
    );
    expect(wrapper.find(".notes__image").exists()).toBe(true);
  });
});
