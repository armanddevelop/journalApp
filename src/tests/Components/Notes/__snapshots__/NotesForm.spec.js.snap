// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in NotesForm Component SnapShot to NotesForm component 1`] = `
<div
  className="notes__content"
>
  <input
    autoComplete="off"
    autoFocus={true}
    className="notes__title-input"
    name="title"
    onChange={[Function]}
    placeholder="Some awsome title"
    type="text"
    value="Hi note"
  />
  <textarea
    className="notes__textarea"
    name="body"
    onChange={[Function]}
    placeholder="What are you doing today"
    value="hi lich"
  />
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete note
  </button>
</div>
`;
